xcodeproj/internal/opts.bzl

```diff
@@ -791,11 +791,11 @@ def _process_compiler_opts(
     has_debug_infos = has_debug_info.keys()
 
     if len(has_debug_infos) == 1:
-        # We don't set "DEBUG_INFORMATION_FORMAT" for "dwarf"-with-dsym",
-        # as that's Xcode's default
         if not has_debug_infos[0]:
             build_settings["DEBUG_INFORMATION_FORMAT"] = ""
-        elif not cpp_fragment.apple_generate_dsym:
+        elif cpp_fragment.apple_generate_dsym:
+            build_settings["DEBUG_INFORMATION_FORMAT"] = "dwarf-with-dsym"
+        else:
             build_settings["DEBUG_INFORMATION_FORMAT"] = "dwarf"
```

Comment:
We can remove this part now, since it's the default. You'll have to adjust the bzl tests. You can change the top comment to:

```
        # We don't set "DEBUG_INFORMATION_FORMAT" for "dwarf", as we set that at
        # the project level.
```